scripts/globals/utils.lua

```diff
@@ -493,6 +493,61 @@ end
 function utils.splitStr(s, sep)
     local fields = {}
     local pattern = string.format("([^%s]+)", sep)
-    string.gsub(s, pattern, function(c) fields[#fields + 1] = c end)
+    local _ = string.gsub(s, pattern, function(c) fields[#fields + 1] = c end)
     return fields
 end
+
+function utils.mobTeleport(mob, hideDuration, pos, disAnim, reapAnim)
```

Comment:
Not going to block this from going in, but there are some things that will need to be considered future forward. I know normally mobs that teleport use a hidden mobskill for that purpose and most of what you had to do here is baked into that. However, there are presently no good examples to give to you (they need cleanup..) and ye old professor may be a special snowflake anyway. We'll figure it out later. Might move this into professor if turns out he's the only one to do this.

Reply:
Angra Mainyu (Dyna Beauc Boss) does something similar to Professor, IE: Randomly teleports to places near his 'spawn' during battle with only an animation.